Demo/src/app/main/main.component.ts

```diff
@@ -1,15 +1,44 @@
-import { Component, OnInit } from '@angular/core';
-
+import { Component } from '@angular/core';
+import {RegService} from "../Service/app.service";
+import {HttpErrorResponse} from "@angular/common/http";
+import {FormGroup, FormControlName, FormControl} from "@angular/forms";
+import{Router} from "@angular/router";
 @Component({
-  selector: 'app-main',
-  templateUrl: './main.component.html',
-  styleUrls: ['./main.component.css']
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
 })
-export class MainComponent implements OnInit {
-
-  constructor() { }
-
-  ngOnInit() {
+export class AppComponent {
+  private myForm:FormGroup;
+ private result:any;
+ constructor(private _service:RegService,
+              private _router:Router){this.myForm= new FormGroup({
+   uname:new FormControl(),
+   ucity:new FormControl(),
+   upwd:new FormControl(),
+   gender:new FormControl(),
+   ustate:new FormControl(),
+   unum:new FormControl()
+ })
+}
+ ngOnInit(){
+   this._service.getData().subscribe(res=>{
+     this.result=res;
+   },(err:HttpErrorResponse)=>{
+     if(err.error instanceof Error){
+       console.log("Client Side Error");
+     }else{
+       console.log("server side error");
+     }
+   });
+  }
+  public Register(){
+   window.localStorage.setItem("name", this.myForm['uname']);
```

Comment:
create localstorage  object and store the data instead of using window.localstorage